tests/shared/src/test/scala/org/http4s/parser/OriginHeaderSuite.scala

```diff
@@ -30,62 +29,55 @@ class OriginHeaderSuite extends munit.FunSuite {
   val hostString2 = "https://127.0.0.1"
 
   test("Origin value method should Render a host with a port number") {
-    val origin: Origin = Origin.HostList(NonEmptyList.of(host1))
+    val origin: Origin = host1
     assertEquals(origin.value, hostString1)
   }
 
   test("Origin value method should Render a host without a port number") {
-    val origin: Origin = Origin.HostList(NonEmptyList.of(host2))
+    val origin: Origin = host2
     assertEquals(origin.value, hostString2)
   }
 
-  test("Origin value method should Render a list of multiple hosts") {
-    val origin: Origin = Origin.HostList(NonEmptyList.of(host1, host2))
-    assertEquals(origin.value, s"$hostString1 $hostString2")
-  }
-
   test("Origin value method should Render an empty origin") {
-    val origin: Origin = Origin.Null
+    val origin: Origin = Origin.`null`
     assertEquals(origin.value, "null")
   }
 
   test("OriginHeader parser should Parse a host with a port number") {
     val text = hostString1
-    val origin = Origin.HostList(NonEmptyList.of(host1))
+    val origin = host1
     val headers = Headers(("Origin", text))
     val extracted = headers.get[Origin]
     assertEquals(extracted, Some(origin))
   }
 
   test("OriginHeader parser should Parse a host without a port number") {
     val text = hostString2
-    val origin = Origin.HostList(NonEmptyList.of(host2))
+    val origin = host2
     val headers = Headers(("Origin", text))
     val extracted = headers.get[Origin]
     assertEquals(extracted, Some(origin))
   }
 
-  test("OriginHeader parser should Parse a list of multiple hosts") {
-    val text = s"$hostString1 $hostString2"
-    val origin = Origin.HostList(NonEmptyList.of(host1, host2))
+  test("OriginHeader parser should Parse a 'null' origin") {
+    val text = "null"
+    val origin = Origin.`null`
     val headers = Headers(("Origin", text))
     val extracted = headers.get[Origin]
     assertEquals(extracted, Some(origin))
   }
 
-  test("OriginHeader parser should Parse an empty origin") {
-    val text = ""
-    val origin = Origin.Null
+  test("OriginHeader should fail on a list of multiple hosts") {
+    val text = s"$hostString1 $hostString2"
     val headers = Headers(("Origin", text))
     val extracted = headers.get[Origin]
-    assertEquals(extracted, Some(origin))
+    assertEquals(extracted, None)
   }
 
-  test("OriginHeader parser should Parse a 'null' origin") {
-    val text = "null"
-    val origin = Origin.Null
+  test("OriginHeader should fail on an empty string") {
```

Comment:
I'm thinking about whether this a bug we should backport, or a breaking change we shouldn't backport.  But it's certainly right for this version.

Reply:
I'd say the former, but I don't think it's super important.

Reply:
That sounds good to me.